src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -672,6 +670,18 @@ public string ApplyGetLogging()
             return _expressionAnonymous;
         }
 
+        internal string ApplyGetLogging(ISanitizedNameProvider nameProvider = null)
+        {
+            if (_expressionAnonymous == null)

```

Comment:
>_expressionAnonymous [](http://example.com/codeflow?start=16&length=20)

We shouldn't reuse _expressionAnonymous cache here. 

Else calling ApplyGetLogging(sanitizer) will change the result of ApplyGetLogging(). 

or even calling: 
ApplyGetLogging(santizer1) 
ApplyGetLogging(santizer2)  // will fail and return sanitizer1 result. 

Probably fine to just not cache here.

Reply:
Can you add a test that illustrates this problem? That will help verify the fix.  :)